src/lib.rs

```diff
@@ -126,12 +126,74 @@ pub struct SelfEncryptor<'a> {
 }
 
 impl<'a> SelfEncryptor<'a> {
-  /// This is the only constructor, for encryptor object
+  /// This is the only constructor, for encryptor object.
   /// Each SelfEncryptor is used for a single file.
   /// The parameters are a DataMap and Storage.
-  /// If new file use DataMap::None as first parameter
+  /// If new file, use DataMap::None as first parameter.
   /// The get and put of Storage need to be implemented to
   /// allow the SelfEncryptor to store encrypted chunks and retrieve when necessary.
+  // # Example usage
+  // ```rust
+  // let mut se = SelfEncryptor::new(&mut my_storage as MyStorage, datamap::DataMap::None);
+  // ```
+  //
+
+/// # Example
+///
+/// SelfEncryptor Example
+///
+/// ```rust
+/// extern crate self_encryption
+/// extern crate rand;
+/// use use self_encryption::*;
+///
+/// fn main() {
```

Comment:
Hi David,
I was thinking of using enum as I wasnt able to push into piece = Vec::<(u
size, String>)::new (). Compiler kept complaining whenever I tried
piece.push(0,"hello). I got the subtract method on a string using
"hello".to_str().as_bytes.iter ().skip ().take ().collect().
But I am more worried converting back to string is an o (n) operation.  I'd
like to use map but that doesn't seem to work
On 22 Mar 2015 20:52, "David Irvine" <notifications@github.com> wrote:

> In src/lib.rs
> <https://github.com/dirvine/self_encryption/pull/17#discussion_r26907043>:
>
> > +  // # Example usage
> > +  // ```rust
> > +  // let mut se = SelfEncryptor::new(&mut my_storage as MyStorage, datamap::DataMap::None);
> > +  // ```
> > +  //
> > +
> > +/// # Example
> > +///
> > +/// SelfEncryptor Example
> > +///
> > +/// ```rust
> > +/// extern crate self_encryption
> > +/// extern crate rand;
> > +/// use use self_encryption::*;
> > +///
> > +/// fn main() {
>
> Good Leo nice to see you getting to grips with this, overwhelming perhaps
> but well worth the effort.
>
> Pair is simple in rust just use (thing1Type, thing2Type). So you can have
> Vec<(thinkg1type, thing2type)> or similar. So tuples everywhere more so
> than c++, but ADT's (enums) in rust also work great if you are only holding
> one type of many options (even with initialisers).
>
> Subtract should also be simple, if you post on slack what you are trying I
> will give you a hand, or post a playpen of it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dirvine/self_encryption/pull/17/files#r26907043>.
>